src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private T[] elements;
+    private int size;
+    private static final int INITIAL_CAPACITY = 10;
+
+    @SuppressWarnings("unchecked")
+    public MateStack() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
     }
 
     public T peek() {
-        return null;
+        throwExeption();
+        return elements[size - 1];
     }
 
     public T pop() {
-        return null;
+        throwExeption();
+        T value = elements[--size];
+        elements[size] = null;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
     }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        T[] newArray = (T[]) new Object[elements.length * 2];
+        System.arraycopy(elements, 0, newArray, 0, elements.length);
+        elements = newArray;
+    }
+
+    private void throwExeption() {
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+    }
```

Comment:
Code Quality Issue: The name of the method 'throwExeption' has a typo. It should be 'throwException'. Also, the method name should reflect its purpose more clearly. For example, 'ensureNotEmpty' would be more descriptive.